plugins/scaffolder/src/components/fields/EntityNamePicker/validation.ts

```diff
@@ -22,6 +22,7 @@ export const entityNamePickerValidation = (
   validation: FieldValidation,
 ) => {
   if (!KubernetesValidatorFunctions.isValidObjectName(value)) {
+    // TODO I need access to i18n object to translate this. Hooks can't be called outside a Component Lifecycle
```

Comment:
@Rugvip I tried to use `translationApiRef` here like the `useTranslationRef` but I got a little confuse If I need to handle the same observable logic from there. Could you give me some direction how to apply here? I can raise I PR after merge this with that change and a update in the docs

Reply:
maybe you could return the plain error string, for example `validation.addError('must_start_and_end_with...')`
and then transform the string to the proper error using the `t` later in the frontend once the error happen

Reply:
I prefer to avoid this @vinzscam, because this push the translation complexity forward.  I will dig when I have some time to provide a simple API here too, like the `useTranslationRef` so we can translate outside react context without any problem (maybe the answer is there and I don't catch yet 😅)